services/port_manager/src/main/java/com/futurewei/alcor/portmanager/service/implement/PortServiceImpl.java

```diff
@@ -146,21 +205,25 @@ public PortWebJson createPort(String projectId, PortWebJson portWebJson) throws
         portEntity.setProjectId(projectId);
 
         try {
-            createPortAsync(portEntity, executor, rollbacks);
+            this.createPortAsync(portEntity, executor, rollbacks, true);
 
             //Wait for all async functions to finish
             List<Object> entities = executor.joinAll();
             entities.add(portEntity);
 
-            //Build GoalState and Send it to DPM
+            NeighborInfo neighborInfo = null;
             if (portEntity.getBindingHostId() != null) {
-                Goalstate.GoalState goalState = GoalStateUtil.buildGoalState(entities, Common.OperationType.CREATE);
+                Map<String, NodeInfo> nodeInfoMap = getNodeInfos(entities);
```

Comment:
Try to use "this.getNodeInfos" if possible.

Reply:
ok